thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detection/algorithm/stage/AnomalyDetectionStageWrapper.java

```diff
@@ -17,46 +17,98 @@
 package com.linkedin.thirdeye.detection.algorithm.stage;
 
 import com.google.common.base.Preconditions;
+import com.linkedin.thirdeye.anomaly.detection.DetectionJobSchedulerUtils;
+import com.linkedin.thirdeye.api.TimeGranularity;
+import com.linkedin.thirdeye.api.TimeSpec;
+import com.linkedin.thirdeye.datalayer.dto.DatasetConfigDTO;
 import com.linkedin.thirdeye.datalayer.dto.DetectionConfigDTO;
 import com.linkedin.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+import com.linkedin.thirdeye.datalayer.dto.MetricConfigDTO;
 import com.linkedin.thirdeye.detection.DataProvider;
 import com.linkedin.thirdeye.detection.DetectionPipeline;
 import com.linkedin.thirdeye.detection.DetectionPipelineResult;
+import com.linkedin.thirdeye.rootcause.impl.MetricEntity;
+import com.linkedin.thirdeye.util.ThirdEyeUtils;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
 import org.apache.commons.collections.MapUtils;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.Interval;
+import org.joda.time.Period;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 
 /**
  * Anomaly Detection Stage Wrapper. This wrapper runs a anomaly detection stage and return the anomalies.
+ * Optionally set the detection window to be moving window fashion. This wrapper will call detection multiple times with
+ * sliding window. Each sliding window start time and end time is aligned to the data granularity. Each window size is set by the spec.
  */
 public class AnomalyDetectionStageWrapper extends DetectionPipeline {
   private static final String PROP_STAGE_CLASSNAME = "stageClassName";
   private static final String PROP_SPECS = "specs";
   private static final String PROP_METRIC_URN = "metricUrn";
 
-  private final AnomalyDetectionStage anomalyDetectionStage;
+  // moving window detection properties
+  private static final String PROP_MOVING_WINDOW_DETECTION = "movingWindowDetection";
+  private static final String PROP_WINDOW_DELAY = "windowDelay";
+  private static final String PROP_WINDOW_DELAY_UNIT = "windowDelayUnit";
+  private static final String PROP_WINDOW_SIZE = "windowSize";
+  private static final String PROP_WINDOW_UNIT = "windowUnit";
+  private static final String PROP_FREQUENCY = "frequency";
+
+  private static final Logger LOG = LoggerFactory.getLogger(AnomalyDetectionStageWrapper.class);
+
   private final String metricUrn;
+  private final Map<String, Object> specs;
+  private final String stageClassName;
+
+  private final int windowDelay;
+  private final TimeUnit windowDelayUnit;
+  private final int windowSize;
+  private final TimeUnit windowUnit;
+  private final boolean movingWindowDetection;
```

Comment:
I'd prefer naming this as isMovingWindowDetection

Reply:
changed to isMovingWindowDetection